### PR TITLE
Harbor dependencies

### PR DESCRIPTION
**Pull request by DeeAjayi**
This PR:

- Documents dependencies needed by Harbor cluster to work properly
- Includes configMap used by postgres as parameters in the helm chart
- Added documentation for creating secret with AWS credentials

### CODE REVIEW

File: DEPENDENCIES.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
+# Dependencies for deploying harbor-operator-app
```
Comment (MarcelMue) [inline DEPENDENCIES.md:1]: Move to docs subfolder maybe?

File: docs/DEPENDENCIES.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+   It is advised to use separate buckets for the WAL backups and the artifacts. Also, the bucket(s) must be accessible by the MC.
+3. AWS Credentials
+   The AWS credentials are needed for the harbor-operator to access the S3 bucket(s). The AWS entity (IAM User, IAM Role etc.) must have the following permissions:
+    - S3:ListBucket
+    - S3:GetObject
+    - S3:PutObject
+    - S3:CreateObject
+    - s3:GetBucket
+4. Subdomain
+    A subdomain is needed for the harbor-operator to create the ingress for the harbor instance. The subdomain must be accessible by the MC.
```
Comment (MarcelMue) [inline docs/DEPENDENCIES.md:18]: Can we clarify this more? AFAIRC we just created an ingress which creates the subdomain

File: docs/DEPENDENCIES.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
+# Dependencies for deploying harbor-operator-app
+## Pre-deployment dependencies
+### Operational requirements
+In order to deploy the harbor-operator-app in a management cluster (MC), you need to have the following available:
+1. S3 Buckets:
```
Comment (MarcelMue) [inline docs/DEPENDENCIES.md:5]: Do these buckets have to be pre-created or will the operators create them with the below permissions?

Comment (DeeAjayi) [reply docs/DEPENDENCIES.md:5]: They have to be pre-created, all the items under the Operational Requirement section have to be pre-created

File: docs/DEPENDENCIES.md (hunk 2 above)
Comment (MarcelMue) [reply docs/DEPENDENCIES.md:18]: I am fine with this PR once we have clarified this.

Comment (cagiti) [reply docs/DEPENDENCIES.md:18]: I know that the harborcluster creates 2 ingress objects, `core` and `notary` (_based on our configuration_). As far as the subdomain is concerned, I'm not sure what creates that. Has someone manually created it, or do you have an operator running somewhere (something like `external-dns`)?

**Review by MarcelMue — APPROVED** [review]

**Review by kubasobon — APPROVED** [review]

**Pull request merged**